### PR TITLE
Install rose in editable mode to avoid symlink issues

### PR DESCRIPTION
**Pull request by wxtim**

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -72,18 +72,21 @@ jobs:
...
           cylc_flow: ${{ github.event.inputs.cylc-flow-tag }}
-          meto_rose: ${{ github.event.inputs.metomi-rose-tag }}
           cylc_rose: ${{ github.event.inputs.cylc-rose-tag }}
           cylc_uis: ${{ github.event.inputs.cylc-uis-tag }}
         run: |
           # NOTE: Install with [all] so we can import plugins which may
           #       have extra dependencies.
           pip install  \
             "cylc-flow[all] @ git+https://github.com/cylc/cylc-flow@${cylc_flow}" \
-            "metomi-rose[all] @ git+https://github.com/metomi/rose@${meto_rose}" \
```
Comment (oliver-sanders) [inline .github/workflows/deploy.yml:83]: Need to keep the meto_rose tag it is important for deployment.

File: src/user-guide/running-workflows/running-workflows.rst [hunk 2]
```diff
@@ -171,10 +171,29 @@ reload are not inserted into the pool automatically. The first instance of each
 must be inserted manually at the right cycle point, with ``cylc insert``.
 
 
+Files Created at Workflow Start
```
Comment (oliver-sanders) [inline src/user-guide/running-workflows/running-workflows.rst:174]: Intended?

Comment (wxtim) [reply src/user-guide/running-workflows/running-workflows.rst:174]: No.

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -78,11 +78,12 @@ jobs:
         run: |
           # NOTE: Install with [all] so we can import plugins which may
           #       have extra dependencies.
-          pip install  \
+          pip install -e \
```
Comment (wxtim) [inline .github/workflows/deploy.yml:81]:
```suggestion
          pip install  \
```

File: .github/workflows/nightly.yml [hunk 4]
```diff
@@ -51,11 +51,12 @@ jobs:
         run: |
           # NOTE: Install with [all] so we can import plugins which may
           #       have extra dependencies.
-          pip install  \
+          pip install -e \
```
Comment (wxtim) [inline .github/workflows/nightly.yml:54]:
```suggestion
          pip install  \
```

File: .github/workflows/test_tutorial_workflow.yml [hunk 5]
```diff
@@ -41,10 +41,11 @@ jobs:
           rose_ref: ${{ github.event.inputs.rose_ref || 'master' }}
           cylc_rose_ref: ${{ github.event.inputs.cylc_rose_ref || 'master' }}
         run: |
-          pip install  \
+          pip install -e \
```
Comment (wxtim) [inline .github/workflows/test_tutorial_workflow.yml:44]:
```suggestion
          pip install  \
```

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -52,11 +52,12 @@ jobs:
         run: |
           # NOTE: Install with [all] so we can import plugins which may
           #       have extra dependencies.
-          pip install  \
+          pip install \
```
Comment (wxtim) [inline .github/workflows/test.yml:55]:
```suggestion
          pip install  \
```

Comment (wxtim) [reply .github/workflows/test.yml]:
```suggestion
          pip install  \
```

File: .github/workflows/deploy.yml [hunk 7]
```diff
@@ -80,9 +80,10 @@ jobs:
           #       have extra dependencies.
           pip install  \
             "cylc-flow[all] @ git+https://github.com/cylc/cylc-flow@${cylc_flow}" \
-            "metomi-rose[all] @ git+https://github.com/metomi/rose@${meto_rose}" \
             "cylc-rose[all] @ git+https://github.com/cylc/cylc-rose@${cylc_rose}" \
             "cylc-uiserver[all] @ git+https://github.com/cylc/cylc-uiserver@${cylc_uis}"
+          git clone --depth 1 https://github.com/metomi/rose ../rose
```
Comment (oliver-sanders) [inline .github/workflows/deploy.yml:85]:
```suggestion
          git clone --depth 1 -b "${meto_rose}" https://github.com/metomi/rose ../rose
```

File: .github/workflows/test.yml [hunk 8]
```diff
@@ -54,9 +54,10 @@ jobs:
           #       have extra dependencies.
           pip install  \
             "cylc-flow[all] @ git+https://github.com/cylc/cylc-flow@${cylc_flow}" \
-            "metomi-rose[all] @ git+https://github.com/metomi/rose@${meto_rose}" \
             "cylc-rose[all] @ git+https://github.com/cylc/cylc-rose@${cylc_rose}" \
             "cylc-uiserver[all] @ git+https://github.com/cylc/cylc-uiserver@${cylc_uis}"
+          git clone --depth 1 https://github.com/metomi/rose ../rose
```
Comment (oliver-sanders) [inline .github/workflows/test.yml:59]:
```suggestion
          git clone --depth 1 -b "${meto_rose}" https://github.com/metomi/rose ../rose
```

File: .github/workflows/deploy.yml [hunk 9]
```diff
@@ -80,9 +80,10 @@ jobs:
...
+          pip install -e ../rose
```
Comment (oliver-sanders) [inline .github/workflows/deploy.yml:86]: Need to pip install `[all]`.

File: .github/workflows/test.yml [hunk 10]
```diff
@@ -54,9 +54,10 @@ jobs:
...
+          pip install -e ../rose
```
Comment (oliver-sanders) [inline .github/workflows/test.yml:60]: Need to pip install `[all]`.

File: .github/workflows/test_tutorial_workflow.yml [hunk 11]
```diff
@@ -43,8 +43,9 @@ jobs:
         run: |
           pip install  \
             "cylc-flow @ git+https://github.com/cylc/cylc-flow@${cylc_ref}" \
-            "metomi-rose @ git+https://github.com/metomi/rose@${rose_ref}" \
             "cylc-rose @ git+https://github.com/cylc/cylc-rose@${cylc_rose_ref}"
+          git clone --depth 1 https://github.com/metomi/rose ../rose
+          pip install -e ../rose
```
Comment (oliver-sanders) [inline .github/workflows/test_tutorial_workflow.yml:48]: Note: **don't** need to pip install `[all]` here.

File: .github/workflows/test_tutorial_workflow.yml [hunk 12]
```diff
@@ -43,8 +43,9 @@ jobs:
         run: |
           pip install  \
             "cylc-flow @ git+https://github.com/cylc/cylc-flow@${cylc_ref}" \
-            "metomi-rose @ git+https://github.com/metomi/rose@${rose_ref}" \
             "cylc-rose @ git+https://github.com/cylc/cylc-rose@${cylc_rose_ref}"
+          git clone --depth 1 https://github.com/metomi/rose ../rose
```
Comment (oliver-sanders) [inline .github/workflows/test_tutorial_workflow.yml:47]:
```suggestion
          git clone --depth 1 -b "${meto_rose}" https://github.com/metomi/rose ../rose
```

File: .github/workflows/nightly.yml [hunk 13]
```diff
@@ -53,9 +53,10 @@ jobs:
           #       have extra dependencies.
           pip install  \
             'cylc-flow[all] @ git+https://github.com/cylc/cylc-flow@master' \
-            'metomi-rose[all] @ git+https://github.com/metomi/rose@master' \
             'cylc-rose[all] @ git+https://github.com/cylc/cylc-rose@master' \
             'cylc-uiserver[all] @ git+https://github.com/cylc/cylc-uiserver@master'
+          git clone --depth 1 https://github.com/metomi/rose ../rose
+          pip install -e ../rose
```
Comment (oliver-sanders) [inline .github/workflows/nightly.yml:59]: Need to pip install `[all]`.

**Review by oliver-sanders — CHANGES_REQUESTED** [review]

File: .github/workflows/test_tutorial_workflow.yml [hunk 14]
```diff
@@ -43,8 +43,9 @@ jobs:
         run: |
           pip install  \
             "cylc-flow @ git+https://github.com/cylc/cylc-flow@${cylc_ref}" \
-            "metomi-rose @ git+https://github.com/metomi/rose@${rose_ref}" \
             "cylc-rose @ git+https://github.com/cylc/cylc-rose@${cylc_rose_ref}"
+          git clone --depth 1 -b "${meto_rose}" https://github.com/metomi/rose ../rose
+          pip install -e ../rose[all]
```
Comment (oliver-sanders) [inline .github/workflows/test_tutorial_workflow.yml:48]:
```suggestion
          pip install -e ../rose
```

Comment (oliver-sanders) [reply .github/workflows/test_tutorial_workflow.yml:48]: https://github.com/cylc/cylc-doc/pull/341#discussion_r756883323

File: .github/workflows/test_tutorial_workflow.yml [hunk 15]
```diff
@@ -38,13 +38,15 @@ jobs:
       - name: install libs
         env:
           cylc_ref: ${{ github.event.inputs.cylc_ref || 'master' }}
+          meto_rose: ${{ github.event.inputs.metomi-rose-tag || 'master' }}
```
Comment (wxtim) [inline .github/workflows/test_tutorial_workflow.yml:41]:
```suggestion
```

File: .github/workflows/test_tutorial_workflow.yml [hunk 16]
```diff
@@ -38,13 +38,15 @@ jobs:
...
           rose_ref: ${{ github.event.inputs.rose_ref || 'master' }}
           cylc_rose_ref: ${{ github.event.inputs.cylc_rose_ref || 'master' }}
         run: |
           pip install  \
             "cylc-flow @ git+https://github.com/cylc/cylc-flow@${cylc_ref}" \
-            "metomi-rose @ git+https://github.com/metomi/rose@${rose_ref}" \
             "cylc-rose @ git+https://github.com/cylc/cylc-rose@${cylc_rose_ref}"
+          git clone --depth 1 --branch "${meto_rose}" https://github.com/metomi/rose ../rose
```
Comment (wxtim) [inline .github/workflows/test_tutorial_workflow.yml:48]:
```suggestion
          git clone --depth 1 --branch "${rose_ref}" https://github.com/metomi/rose ../rose
```

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**